pkg/repository/workspace.go

```diff
@@ -0,0 +1,87 @@
+package repository
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"time"
+
+	"github.com/beam-cloud/beta9/pkg/common"
+)
+
+type WorkspaceRedisRepository struct {
+	rdb *common.RedisClient
+}
+
+func NewWorkspaceRedisRepository(r *common.RedisClient) WorkspaceRepository {
+	return &WorkspaceRedisRepository{rdb: r}
+}
+
+type VolumePathTokenData struct {
+	WorkspaceId string `redis:"workspace_id"`
+	VolumePath  string `redis:"volume_path"`
+}
+
+func (wr *WorkspaceRedisRepository) ValidateWorkspaceVolumePathDownloadToken(workspaceId string, volumePath string, token string) error {
+	key := common.RedisKeys.WorkspaceVolumePathDownloadToken(token)
+
+	var data VolumePathTokenData
+	res, err := wr.rdb.HGetAll(context.TODO(), key).Result()
+	if err != nil {
+		return err
+	}
+
+	if len(res) == 0 {
+		return errors.New("invalid token")
+	}
+
+	err = common.ToStruct(res, &data)
+	if err != nil {
+		return err
+	}
+
+	if data.WorkspaceId != workspaceId || data.VolumePath != volumePath {
+		return errors.New("invalid token")
+	}
+
+	return nil
+}
+
+func generateToken(length int) (string, error) {
```

Comment:
UUID might be simpler. Its random (uuid4) without having to deal with returning an error.

Reply:
Either is fine, just suggesting a simpler way to generate a random token. Feel free to keep this as-is.